packages/trader/src/AppV2/Components/TradeParameters/Stake/stake.tsx

```diff
@@ -53,6 +53,16 @@ const Stake = observer(({ is_minimized }: TTradeParametersProps) => {
         ? Number(v2_params_initial_values.stake)
         : available_contract_types?.[contract_type]?.config?.default_stake;
 
+    useEffect(() => {
+        if (client_currency !== currency) {
+            onChange({ target: { name: 'currency', value: client_currency } });
+            if (!isCryptocurrency(client_currency ?? '')) {
+                onChange({ target: { name: 'amount', value: default_stake } });
+                setV2ParamsInitialValues({ value: default_stake as number, name: 'stake' });
+            }
+        }
+    }, [client_currency]);
```

Comment:
Should `currency` also be in array with dep-s?

Reply:
I am just keeping a check with the client_currency so that  whenever client_currency updates , I am syncing it with the currency . I don't see a case where in a logged_in user , I need to check on currency change as it is working as expected .